packages/gatsby-script/src/gatsby-script.tsx

```diff
@@ -120,7 +133,40 @@ function injectScript(props: ScriptProps): HTMLScriptElement | null {
     onError,
   } = props || {}
 
-  if (scriptCache.has(id || src)) {
+  const scriptKey = id || src || (scriptCache.size + 1).toString()
```

Comment:
I don't understand `(scriptCache.size + 1).toString()` part - what is it meant to do?

This seems to me like this would be hit for inline scripts that don't have ids? Is there any other case we would hit it?

Reply:
Chatted a bit with Ty on this outside of GitHub. This was needed to make sure we actually add multiple inline scripts (it seems). While this is not ideal, I will merge this PR in and some refactor should happen in follow up so we don't add "random" numbers like that to cache, because it will potentially will grow uncontrollably over time.